spaceship/lib/spaceship/tunes/iap.rb

```diff
@@ -75,6 +75,17 @@ def create!(type: "consumable",
                            family_id: family_id,
                            subscription_duration: subscription_duration,
                            subscription_free_trial: subscription_free_trial)
+
+        # Update pricing for a recurring subscription.
+        if type == Spaceship::Tunes::IAPType::RECURRING && pricing_intervals
+          # There are cases where the product that was just created is not immediately found,
+          # and in order to update its pricing the purchase_id is needed. Therefore polling is done
+          # until it is found.
+          product = find(product_id) until product
```

Comment:
I don't know the codebase, but are there other parts of the code that use the same method for similar problems?

Reply:
Not that I know of, which sounds weird to me because creating a product and updating it immediately afterwards seems like a common scenario.
Another solution I came up with is to sleep for a few seconds. I tried it, but the amount needed to sleep might change from one connection to another. Note that `find(product_id)` itself takes a second or so, so there won't be so many URL requests and the polling isn't that consuming.
BTW just wanted to make sure: I assumed that if the product's creation failed an exception will be raised (otherwise the polling will be stuck forever).

Reply:
No idea, I know nothing about that part of the codebase, but that was one of the things I was wondering as well. Or what happens if the product just doesn't appear because of whatever.

Reply:
You can see in `handle_itc_response` that if there was an error an exception is raised. I created a few dozen products with the new Spaceship API already and it never got stuck: I always got an exception or it was created successfully. Do you think I should add a timeout?

Reply:
Nah I think this is good until it really breaks because of that. No unneeded complexity - this is all hard enough already.

Reply:
If this method is called with a nonexistent `product_id`, wouldn't this result in an infinite loop since the product is always not there?

Reply:
What do you mean? This method creates a new product with `product_id`. From what I saw, every time the creation failed an exception was raised. Though I agree that a retry counter should be here just in case.

Reply:
Consider this for example:

* Process A creates product
* Process B deletes same product
* Process A hits this line and hangs up

Reply:
You will run into an exception if you try to create a product with the same `product_id`, even if the product was deleted. Anyway, I will add a retry counter.

Reply:
I didn't say that two products were created at all. All process A does it to try to create the product with the lines above this one, and while Process A is still in the same execution of this method (or suspended inbetween due to high load), process B deletes the product. Then process A would hit an infinite loop.

Reply:
Right, in this case process A will hit an infinite loop.